diff --git a/src/components/ChatWindow.tsx b/src/components/ChatWindow.tsx
index 068fc4dddd..02d00d255c 100644
--- a/src/components/ChatWindow.tsx
+++ b/src/components/ChatWindow.tsx
@@ -91,6 +91,11 @@ const ChatMessage = ({ message }: { message: Message }) => {
         {getMessageIcon(message)}
       </div>
       <span className="mr-2 font-bold">{getMessagePrefix(message)}</span>
+      {message.type == "thinking" && (
+        <span className="italic text-zinc-400">
+          (Restart if this takes more than 30 seconds)
+        </span>
+      )}
       <span>{message.value}</span>
     </div>
   );
