diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 479d960440..0975d6f541 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -105,7 +105,7 @@ const Home: NextPage = () => {
                 value={name}
                 disabled={agent != null}
                 onChange={(e) => setName(e.target.value)}
-                placeholder="AgentGPT (Note: this field doesn't do anything right now)"
+                placeholder="AgentGPT"
               />
 
               <Input
