diff --git a/src/pages/index.tsx b/src/pages/index.tsx
index 932d78ce4d..12302d0cec 100644
--- a/src/pages/index.tsx
+++ b/src/pages/index.tsx
@@ -32,13 +32,15 @@ const Home: NextPage = () => {
     const savedModalData = localStorage.getItem(key);
 
     // Momentarily always run
-    if (true) {
-      setTimeout(() => {
+    setTimeout(() => {
+      if (savedModalData == null) {
+        setShowHelpDialog(true);
+      } else {
         setShowSettingsDialog(true);
-      }, 1500);
-    }
+      }
+    }, 1500);
 
-    // localStorage.setItem(key, JSON.stringify(true));
+    localStorage.setItem(key, JSON.stringify(true));
   }, []);
 
   useEffect(() => {
